Source/loadsave.cpp

```diff
@@ -207,313 +207,309 @@ class SaveHelper {
+	player.position.offset2.deltaX = file.NextLE<int32_t>();
+	player.position.offset2.deltaY = file.NextLE<int32_t>();
+	file.Skip(4); // Skip actionFrame
 	for (uint8_t i = 0; i < giNumberOfLevels; i++)
-		player._pLvlVisited[i] = file->NextBool8();
+		player._pLvlVisited[i] = file.NextBool8();
 	for (uint8_t i = 0; i < giNumberOfLevels; i++)
-		player._pSLvlVisited[i] = file->NextBool8();
-
-	file->Skip(2);     // Alignment
-	file->Skip(4);     // skip _pGFXLoad
-	file->Skip(4 * 8); // Skip pointers _pNAnim
-	player._pNFrames = file->NextLE<int32_t>();
-	file->Skip(4);     // skip _pNWidth
-	file->Skip(4 * 8); // Skip pointers _pWAnim
-	player._pWFrames = file->NextLE<int32_t>();
-	file->Skip(4);     // skip _pWWidth
-	file->Skip(4 * 8); // Skip pointers _pAAnim
-	player._pAFrames = file->NextLE<int32_t>();
-	file->Skip(4); // skip _pAWidth
-	player._pAFNum = file->NextLE<int32_t>();
-	file->Skip(4 * 8); // Skip pointers _pLAnim
-	file->Skip(4 * 8); // Skip pointers _pFAnim
-	file->Skip(4 * 8); // Skip pointers _pTAnim
-	player._pSFrames = file->NextLE<int32_t>();
-	file->Skip(4); // skip _pSWidth
-	player._pSFNum = file->NextLE<int32_t>();
-	file->Skip(4 * 8); // Skip pointers _pHAnim
-	player._pHFrames = file->NextLE<int32_t>();
-	file->Skip(4);     // skip _pHWidth
-	file->Skip(4 * 8); // Skip pointers _pDAnim
-	player._pDFrames = file->NextLE<int32_t>();
-	file->Skip(4);     // skip _pDWidth
-	file->Skip(4 * 8); // Skip pointers _pBAnim
-	player._pBFrames = file->NextLE<int32_t>();
-	file->Skip(4); // skip _pBWidth
-
-	LoadItems(file, NUM_INVLOC, player.InvBody);
-	LoadItems(file, NUM_INV_GRID_ELEM, player.InvList);
-	player._pNumInv = file->NextLE<int32_t>();
+		player._pSLvlVisited[i] = file.NextBool8();
+
+	file.Skip(2);     // Alignment
+	file.Skip(4);     // skip _pGFXLoad
+	file.Skip(4 * 8); // Skip pointers _pNAnim
+	player._pNFrames = file.NextLE<int32_t>();
+	file.Skip(4);     // skip _pNWidth
+	file.Skip(4 * 8); // Skip pointers _pWAnim
+	player._pWFrames = file.NextLE<int32_t>();
+	file.Skip(4);     // skip _pWWidth
+	file.Skip(4 * 8); // Skip pointers _pAAnim
+	player._pAFrames = file.NextLE<int32_t>();
+	file.Skip(4); // skip _pAWidth
+	player._pAFNum = file.NextLE<int32_t>();
+	file.Skip(4 * 8); // Skip pointers _pLAnim
+	file.Skip(4 * 8); // Skip pointers _pFAnim
+	file.Skip(4 * 8); // Skip pointers _pTAnim
+	player._pSFrames = file.NextLE<int32_t>();
+	file.Skip(4); // skip _pSWidth
+	player._pSFNum = file.NextLE<int32_t>();
+	file.Skip(4 * 8); // Skip pointers _pHAnim
+	player._pHFrames = file.NextLE<int32_t>();
+	file.Skip(4);     // skip _pHWidth
+	file.Skip(4 * 8); // Skip pointers _pDAnim
+	player._pDFrames = file.NextLE<int32_t>();
+	file.Skip(4);     // skip _pDWidth
+	file.Skip(4 * 8); // Skip pointers _pBAnim
+	player._pBFrames = file.NextLE<int32_t>();
+	file.Skip(4); // skip _pBWidth
+
+	for (Item &item : player.InvBody)
+		LoadItemData(file, item);
+	for (Item &item : player.InvList)
+		LoadItemData(file, item);

```

Comment:
Consider leaving an empty line after these scopeless statements for readability.